MoneyAPP/moneyapp/lib/pages/loan_page.dart

```diff
@@ -13,8 +13,10 @@ class LoanPage extends StatefulWidget {
 
 class LoanPageState extends State<LoanPage> {
   final _formKey = GlobalKey<FormState>();
-  final TextEditingController _monthlySalaryController = TextEditingController();
-  final TextEditingController _monthlyExpensesController = TextEditingController();
+  final TextEditingController _monthlySalaryController =
+      TextEditingController();
+  final TextEditingController _monthlyExpensesController =
+      TextEditingController();
```

Comment:
Now, lets go into the next task. I want you to introduce to `flutter_hooks` as it is used quite a lot in our code.

The state management is use to not have to deal with Flutter's State class, like you are doing here. We use BLoC/Cubit to extract the business logic from the UI into another class, and it is also separable by semantics, meaning we can have 2 Cubits doing entire different thing, but still being used by the same piece of UI code.

By doing that, we don't want to have any States in the app left, like this `LoanPageState`. But not all of the code should be in the BLoC. This TextEditingControllers could be in the State of a Cubit, but it feels unnatural to keep them there, as they have their own state inside them. When text changes, it is the same `TextEditingController`, and it doesn't make sense to update BLoC/Cubit state because it will still be the same, yet we would need to to update the UI.

To remedy this, we often use flutter_hooks:
- [ ] implement it to pubspec.yaml
- [ ] Convert this LoanPage to HookWidget, and make sure the build method is there (remove the LoanPageState class)
- [ ] put the line `final _monthlySalaryController = useTextEditingController()` in the actual build() method
- [ ] use the _monthlySalaryController as you use it now.

You need to read how hooks work if you didn't work with them yet. When build() method is called first time, it will create new hooks for every useSomething, but next time, it will just return the old ones, by the same order. It is important to not have any logic with calling the hooks, to make sure they are same order every time they are called.